### PR TITLE
Add ability to escape commas and braces in transformer properties

### PR DESCRIPTION
**Pull request by AlenaBudevich**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTable.java [hunk 1]
```diff
@@ -219,7 +219,8 @@ private String stripProperties(String headerSeparator, String valueSeparator, St
         Matcher matcher = INLINED_PROPERTIES_PATTERN.matcher(tableWithoutProperties);
         while (matcher.matches()) {
             String propertiesAsString = matcher.group(1);
-            propertiesList.add(new ExamplesTableProperties(propertiesAsString, headerSeparator, valueSeparator, ignorableSeparator));
+            propertiesList.add(new ExamplesTableProperties(propertiesAsString.replaceAll("\\\\\\{", "{")
+                    .replaceAll("\\\\}", "}"), headerSeparator, valueSeparator, ignorableSeparator));
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTable.java]: `org.apache.commons.lang3.StringUtils.replace(String, String, String)`

File: jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTableProperties.java [hunk 2]
```diff
@@ -57,9 +58,11 @@ public ExamplesTableProperties(String propertiesAsString, String defaultHeaderSe
 
     private Map<String, String> parseProperties(String propertiesAsString) {
         Map<String, String> result = new LinkedHashMap<>();
-        for (String propertyAsString : StringUtils.split(propertiesAsString, ',')) {
-            String[] property = StringUtils.split(propertyAsString, "=", 2);
-            result.put(property[0].trim(), property[1].trim());
+        if (!isBlank(propertiesAsString)) {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTableProperties.java]: why is this check required?

File: jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTableProperties.java [hunk 3]
```diff
@@ -57,9 +58,11 @@ public ExamplesTableProperties(String propertiesAsString, String defaultHeaderSe
...
+            for (String propertyAsString : propertiesAsString.split("(?<!\\\\),")) {
+                String[] property = StringUtils.split(propertyAsString, "=", 2);
+                result.put(property[0].trim(), property[1].replaceAll("\\\\,", ",").trim());
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTableProperties.java]: `org.apache.commons.lang3.StringUtils.replace(String, String, String)`

File: jbehave-core/src/test/java/org/jbehave/core/model/ExamplesTablePropertiesBehaviour.java [hunk 4]
```diff
@@ -27,6 +28,8 @@ public void canGetCustomProperties() {
         assertThat(properties.isTrim(), is(false));
         assertThat(properties.isMetaByRow(), is(true));
         assertThat(properties.getTransformer(), equalTo("CUSTOM_TRANSFORMER"));
+        assertThat(properties.getProperties().getProperty("tables"),
+                equalTo("{transformer=CUSTOM_TRANSFORMER, parameter1=value1}"));
```
Comment (valfirst) [inline jbehave-core/src/test/java/org/jbehave/core/model/ExamplesTablePropertiesBehaviour.java]: add a new test, do not change existing ones

File: jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTableProperties.java (hunk 2 above)
Comment (AlenaBudevich) [reply jbehave-core/src/main/java/org/jbehave/core/model/ExamplesTableProperties.java]: `StringUtils.split` and `split()` of String work different ways

File: distribution/src/site/content/tabular-parameters.html [hunk 5]
```diff
@@ -267,6 +267,11 @@ <h2>Using table transformers</h2>
                 .useStoryParser(new RegexStoryParser(tableFactory))
 ]]>
 </script>
+In addition, there is the possibility to escape commas and braces in transformer properties:
+<pre class="brush: plain">
+{transformer=CUSTOM_TRANSFORMER, table=\{name=table1\, parameter1=value1\}}
```
Comment (valfirst) [inline distribution/src/site/content/tabular-parameters.html]: {propertyKey=\{innerKey1=innerValue1\, innerKey2=innerValue2\}}

**Pull request merged**